src/com/nutrons/FRamework/StreamManager.java

```diff
@@ -0,0 +1,58 @@
+package com.nutrons.FRamework;
+
+import io.reactivex.Flowable;
+import io.reactivex.Observable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class sets up the I/O factories and initializes subsystems.
+ */
+public class StreamManager {
+  private List<Subsystem> subsystems;
+  private Flowable<Boolean> enabled;
+  private Flowable<CompMode> mode;
+
+  /**
+   * Manages subsystems and the competition loop.
+   * @param enabled a Flowable of booleans, representing changes in the enabled state of the robot
+   * @param mode a Flowable of CompModes, representing changes in the competition game mode
+   */
+  public StreamManager(Flowable<Boolean> enabled, Flowable<CompMode> mode) {
+    this.subsystems = new ArrayList<>();
+    this.enabled = enabled;
+    this.mode = mode;
+  }
+
+  /**
+   * Called to subscribe subsystem streams, and start the competition loop.
+   */
+  public void startCompetition() {
+    Observable.fromIterable(this.subsystems).blockingSubscribe(Subsystem::registerSubscriptions);
```

Comment:
You might want them to subscribe when `registerSubsystem` is called, otherwise people might be confused when things don't update if they are added later.  Alternately, make this **SUPER EXPLICIT** in the documentation